src/components/pages/doc.js

```diff
@@ -258,7 +258,8 @@ function Doc({ initialState, PageTemplate }) {
         { state.revisions.length > 0 && revisionPicker}
         <DocView
           documentOrRevision={state.currentDocOrRevision}
-          onAction={isHardDeleteEnabled ? handleAction : null}
+          disabledActions={disabledSectionActions}
```

Comment:
Separating discussion in second thread.

**laurentiu-ilici**
This construct seems backwards to me. I had to dig down 3 files from here to understand why you send "disabledActions" instead of "possible/enabledActions".

**adrianaluchian**
`disabledActions` is because that is the exception from the rule, and the rule is that all actions are enabled.

**laurentiu-ilici**
To me it seems weird that the logic for what is enables / disables the actions now lives in 2 places:
doc.js has: const disabledSectionActions = [...isExternalDocument ? [SECTION_ACTIONS.hardDelete] : []];
section-display: in between that just passes the disabledActions
section-action-dropdown: again knows about all actions and adds some additional restraints on this particular action:
if (canHardDelete && !section.deletedOn && //the rest)

Would it not be more streamlined if the
section-action-dropdown just knew how to display in stuff as a dumb component that receives it's actions from someone who knows?
Alternatively, it could keep all the conditions. It's the split that is a bit hard to maintain I think.

Reply:
What I meant by "the rule is that all actions are enabled" was that the components involved are reused in some cases, and I did not want to escalate into a big change with risk of side-bugs.

`section-action-dropdown` is already not a dumb component, given it decides `canHardDelete`. I agree that the conditions to show the delete menu come from 2 different levels. To me they should come from above, but I would not want to escalate into a refactoring.
So I will then change to pass `isExternalDocument` from the page down through the components instead of `disabledActions` and have the lowest component decide `canHardDelete` based on this, to keep the pattern.

We can definitely rethink these components if we want different modes from them and we will probably have to for introducing inline editing of documents.

Reply:
Actually scratch that, since the condition to disable editing will no longer be `isExternalDocument` but `state.type === DOCUMENT_TYPE.revision && isExternalDocument` 😂 I wouldn't pass the document type to the lower components in this case... I'd rather keep `disabledActions`

Reply:
😆

Reply:
Got one better:
I gave up fighting to keep the `onAction` generic and went back to not attaching an event handler to it at all when section deletion is disabled. Otherwise there were more changes to make so that the UI doesn't show a border on hovering the sections when `doc` page was not in `review` mode.

We'll redo this part with `doc` inline edit.

Please check the changes once more that I haven't missed something 🙏

Reply:
Ah, I checked before and I saw that the border was there but decided to ignore it.
I'll check.

Reply:
But Andreas pointed it out to me earlier and from then on I could no longer unsee it :D